lib/pages/account_creation.dart

```diff
@@ -69,18 +72,28 @@ class _AccountCreationState extends State<AccountCreation> {
                     else
                       return null;
                   },
-                  onSaved: (value) => password = value,
+                  onSaved: (value) => _password = value,
                 ),
               ),
               Container(
                 padding: EdgeInsets.only(left: 40.0, right: 40.0, top: 5.0),
                 child: RaisedButton(
-                  onPressed: () {
+                  onPressed: () async {
                     if (_accountKey.currentState.validate()) {
                       _accountKey.currentState.save();
 
                       //Call Firebase to create Authenticated User.
-                      accountService.saveAccount(email, name, password);
+                      try {
+                        FirebaseUser result = await Provider.of<AuthService>(context).createUser(email: _email, password: _password);
+                        print(result);
+                        Navigator.pushReplacementNamed(context, '/home');
+                      } on AuthException catch (error) {
+                        // handle the firebase specific error
+                        return _buildErrorDialog(context, error.message);
```

Comment:
Don't you need an `await` here? I think otherwise it wall fall out before the dialog error pops up

Reply:
There is an await on calling Provider.of<AuthService>(context).createUser(email: _email, password: _password); Therefore before calling Navigator.push() we ensure no exceptions are thrown. I tested by force throwing an exception and the error dialog seemed to be working fine.